lib/xchain/merkle.go

```diff
@@ -7,6 +7,11 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
+const (
+	DomainSeparationTagBlockHeader merkle.DomainSeparationTag = 1 + iota
+	DomainSeparationTagMessage
```

Comment:
lets explicitly number these enums. I suggest to only use `iota` for in-memory enums. When persisting enums to disk, it is better to be super explicit about there values, this avoid accidental changing of values due reordering or deprecation. Also improves lookup. Don't need to count in your head.

TL;DR: only use iota if the actual number doesn't matter.

Reply:
Can maybe make names more concise, especially since the type is already full qualified.
```
const (
    DSTUnknown  merkle.DomainSeparationTag = 0
    DSTBlockHeader merkle.DomainSeparationTag = 1
    DSTMessage merkle.DomainSeparationTag = 2
)
```